circleci.go

```diff
@@ -1015,10 +1016,15 @@ func (c *Client) TriggerPipelineWithContext(ctx context.Context, vcsType VcsType
 	if c.Version < APIVersion2 {
```

Comment:
GH won't let me comment above, but let's fix the godoc removing "It is not exactly specified in the documentation
// which takes precendence when both are specified." with saying "It will throw an error if both branch and tag are specified".